bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/image/ImageCatalogClient.scala

```diff
@@ -0,0 +1,266 @@
+package edu.gemini.catalog.image
+
+import java.io._
+import java.net.URL
+import java.nio.file.{Files, Path, StandardCopyOption}
+import java.util.logging.Logger
+
+import edu.gemini.spModel.core.{Angle, Coordinates, Declination, RightAscension}
+import nom.tam.fits.{Fits, Header}
+
+import scala.util.matching.Regex
+import scala.math._
+import scala.reflect.ClassTag
+import scalaz.Scalaz._
+import scalaz._
+import scalaz.concurrent.Task
+
+/**
+  * Angular size of an image. Used to store the real size of images instead to the requested size
+  */
+case class AngularSize(ra: Angle, dec: Angle) {
+  def this(size: Angle) = this(size, size)
+
+  def halfDec = (dec / 2).getOrElse(Angle.zero)
+  def halfRa = (ra / 2).getOrElse(Angle.zero)
+}
+
+case object AngularSize {
+  val zero = AngularSize(Angle.zero, Angle.zero)
+
+  /** @group Typeclass Instances */
+  implicit val equal = Equal.equalA[AngularSize]
+
+  /** @group Typeclass Instances */
+  implicit val order: Order[AngularSize] =
+    Order.orderBy(a => (a.ra.toRadians * a.dec.toRadians))
```

Comment:
Good catch. I struggled with this, and it was `(ra, dec)` for a while. You're right that `equals` would be violated but it didn't seem right that you'd consider an image with ra small but a large dec smaller than one with ra slightly bigger but dec small.

This is mostly academic since it is only used to discern what image to pick if the angular distance from target to center is the same between 2 images, and images are mostly equal size for a given catalog. In fact this is only a problem in tests

I'll change it as suggested